api/test/unit/utilities/lottery-date-validator.spec.ts

```diff
@@ -0,0 +1,172 @@
+import {
+  ListingEventsTypeEnum,
+  ReviewOrderTypeEnum,
+  YesNoEnum,
+} from '@prisma/client';
+import { LotteryDateParamValidator } from '../../../src/utilities/lottery-date-validator';
+
+describe('Testing OrderQueryParamValidator', () => {
+  it('should return true if reviewOrderType is not lottery and lotteryOptIn is null', () => {
+    const lotteryDateParamValidator = new LotteryDateParamValidator();
+    expect(
+      lotteryDateParamValidator.validate([], {
+        property: 'listingEvents',
+        object: {
+          reviewOrderType: ReviewOrderTypeEnum.firstComeFirstServe,
+          lotteryOptIn: null,
+        },
+        value: undefined,
+        constraints: [],
+        targetName: '',
+      }),
+    ).toBeTruthy();
+  });
+
+  it('should return true if reviewOrderType is lottery and lotteryOptIn is no', () => {
+    const lotteryDateParamValidator = new LotteryDateParamValidator();
+    expect(
+      lotteryDateParamValidator.validate([], {
+        property: 'listingEvents',
+        object: {
+          reviewOrderType: ReviewOrderTypeEnum.lottery,
+          lotteryOptIn: false,
+        },
+        value: undefined,
+        constraints: [],
+        targetName: '',
+      }),
+    ).toBeTruthy();
+  });
+
+  it('should return false if reviewOrderType is lottery and lotteryOptIn is yes and listingEvents is empty', () => {
+    const lotteryDateParamValidator = new LotteryDateParamValidator();
+    expect(
+      lotteryDateParamValidator.validate([], {
+        property: 'listingEvents',
+        object: {
+          reviewOrderType: ReviewOrderTypeEnum.lottery,
+          lotteryOptIn: true,
+        },
+        value: undefined,
+        constraints: [],
+        targetName: '',
+      }),
+    ).toBeFalsy();
+  });
+
+  it('should return true if reviewOrderType is lottery and lotteryOptIn is yes and listingEvents has publicLottery without startDate', () => {
```

Comment:
This says it should return 'true', but the test is validating that it's 'falsy'. Same with the two tests below this as well

Reply:
Oops, some copy 🍝